cpu/stm32_common/periph/pm.c

```diff
@@ -49,13 +61,21 @@ void pm_set(unsigned mode)
             /* Clear PDDS and LPDS bits to enter stop mode on */
             /* deepsleep with voltage regulator on */
             PWR->CR &= ~(PWR_CR_PDDS | PWR_CR_LPDS);
+            PWR->CR |= PM_STOP_CONFIG;
             /* Set SLEEPDEEP bit of system control block */
             deep = 1;
             break;
     }
 #endif
 
     cortexm_sleep(deep);
+
+#if defined(CPU_FAM_STM32F1) || defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4)
```

Comment:
I ran another test.

With the current code, it works as @kaspar030 pointed out: another thread can be scheduled before PLL is reinitialized.

If we change `cortexm_sleep()` from 
```C
static inline void cortexm_sleep(int deep)
{
    if (deep) {
        SCB->SCR |=  (SCB_SCR_SLEEPDEEP_Msk);
    }
    else {
        SCB->SCR &= ~(SCB_SCR_SLEEPDEEP_Msk);
    }

    /* ensure that all memory accesses have completed and trigger sleeping */
    __disable_irq();
    __DSB();
    __WFI();
    __enable_irq();
}
```
to:
```C
static inline void cortexm_sleep(int deep)
{
    if (deep) {
        SCB->SCR |=  (SCB_SCR_SLEEPDEEP_Msk);
    }
    else {
        SCB->SCR &= ~(SCB_SCR_SLEEPDEEP_Msk);
    }

    /* ensure that all memory accesses have completed and trigger sleeping */
    unsigned state = irq_disable();
    __DSB();
    __WFI();
    irq_restore(state);
}
```
i.e. replacing `__disable_irq()/__enable_irq()` by `irq_disable()/irq_restore(state)`.
Then the issue is no longer present: the clock is initialized, then the irq are restored and the interrupt is served, then only after threads can be scheduled.

So there are 2 choices: either we change `cortexm_sleep` as proposed here, or we remove disabling/re-enabling interrupt from `cortexm_sleep` and let the layers above taking care of interrupts.

@kaspar030 @gebart @haukepetersen what's your opinion on that?